src/libData/BlockData/BlockHeader/BlockHeaderBase.cpp

```diff
@@ -38,6 +42,35 @@ BlockHash BlockHeaderBase::GetMyHash() const {
   return blockHash;
 }
 
+const uint32_t& BlockHeaderBase::GetVersion() const { return m_version; }
+
+void BlockHeaderBase::SetVersion(const uint32_t& version) {
+  m_version = version;
+}
+
 const CommitteeHash& BlockHeaderBase::GetCommitteeHash() const {
   return m_committeeHash;
 }
+
+void BlockHeaderBase::SetCommitteeHash(const CommitteeHash& committeeHash) {
+  m_committeeHash = committeeHash;
+}
+
+const BlockHash& BlockHeaderBase::GetPrevHash() const { return m_prevHash; }
+
+void BlockHeaderBase::SetPrevHash(const BlockHash& prevHash) {
+  m_prevHash = prevHash;
+}
+
+bool BlockHeaderBase::operator==(const BlockHeaderBase& header) const {
+  return std::tie(m_version, m_committeeHash, m_prevHash) ==
+         std::tie(header.m_version, header.m_committeeHash, header.m_prevHash);
+}
+
+bool BlockHeaderBase::operator<(const BlockHeaderBase& header) const {
+  return m_version < header.m_version;
```

Comment:
Is this check sufficient?

Reply:
yeah as there is no other usage of this operator currently, should be enough as the hash shouldn't be used for comparison anyway.